### PR TITLE
Allow users to a use optimizers that are subclass of torch.optim.Optimizers

### LINKED ISSUE #53: Support for arbitrary optimizer
**Is your feature request related to a problem? Please describe.**
Not exactly a problem, but it would be nice to have support for optimizers other than just Adam and AdamW.

**Describe the solution you'd like**
Give user the flexibility to choose his/her own choice of optimizer as long as it is inherited from [`torch.optim.Optimizer`](https://pytorch.org/docs/stable/optim.html#torch.optim.Optimizer).

### PR DESCRIPTION
**Pull request by prakhar6sharma**
Fixes #53. Supports all optimizers supported by [`torch.optim.Optimizers`](https://pytorch.org/docs/stable/optim.html#algorithms) except `SGD` and `LBFGS`.

`SGD` requires `lr` to be a fixed argument but the current configure arguments doesn't allow that as we are using a LRScheduler. `LBFGS` doesn't support per-parameter options but currently configure arguments uses variable weight decay.

Currently, I am working on writing tests for this PR. Also, I would update the notebooks in the coming commits as the API has changed and it no longer takes optimizer as string but takes it as a Callable class. 

To Do:
- [x] Update notebooks to reflect the API Change
- [x] Write tests

### CODE REVIEW

**Comment by jasonjewik** [comment]
Would you mind updating the documentation too? We should explain how to select an optimizer on the [Models page](https://climatelearn.readthedocs.io/en/latest/user-guide/models.html#).

If you do not know how to update the documentation, that's OK (I haven't written that section of our docs yet). You can provide examples in a comment here or in the example notebooks, and I'll update the docs on your behalf.

**Comment by prakhar6sharma** [comment]
@jasonjewik, I can add the docs later as a separate PR. I have added basic tests for now. In later versions, we can discuss more about how to come up with better tests.

I think this PR is ready to be merged now.

File: src/climate_learn/models/modules/forecast.py [hunk 1]
```diff
@@ -10,12 +10,14 @@
 from .utils.lr_scheduler import LinearWarmupCosineAnnealingLR
 from .utils.metrics import lat_weighted_acc, lat_weighted_mse, lat_weighted_rmse
 
+OptimizerCallable = Callable[[Iterable], torch.optim.Optimizer]
```
Comment (jasonjewik) [inline src/climate_learn/models/modules/forecast.py:13]: Can you explain why we have `Iterable` here?

Comment (prakhar6sharma) [reply src/climate_learn/models/modules/forecast.py:13]: I don't know exactly why. I referred it from [here](https://pytorch-lightning.readthedocs.io/en/latest/cli/lightning_cli_advanced_3.html#multiple-optimizers-and-schedulers).

Comment (jasonjewik) [reply src/climate_learn/models/modules/forecast.py:13]: OK, sgtm then.
